### DISCUSSION TITLE
'Sample point not found' in VectorPostprocessor

### DISCUSSION THREAD

**Discussion by tsypik** [idaholab/moose#23041] [Q&A General] [answered]
I am using PointValueSampler to extract values at the top of the mesh. I am reading points from an XYZ file, that corresponds to a surface, the same one I used for meshing. 
``````
 [VectorPostprocessors]
  [pressure_at_top_quaternary]
    type = GolemPointValueSamplerFromFile
    file_name = 'data/Z_quaternary.txt'
    dimension = 3
    contains_complete_history = true
    sort_by = 'z'
    variable = 'pore_pressure'
    outputs = 'p_quaternary_out'
    execute_on = 'FINAL'
[]
``````
This operation ends with an error: 
```
sample point not found: (x,y,z)=(4.45201e+06, 5.72e+06,  314.154)
````
I think the issue is due to the rounding of coordinates, which forces a sample point that lies on the boundary to end up outside of the mesh. Any idea how to overcome this without utilizing NodalValueSampler or SideValueSampler?

**Comment by GiudGiud** [comment, marked as answer]
Hello

I suggest you round the coordinates manually inside the mesh. There s no parameters in the PointValueSampler to do that
Nodal and SideValueSampler are good ideas as well

Is the surface flat? You could use a planevaluesampler right behind it

Guillaume
